pulpcore/app/models/repository.py

```
@@ -379,7 +379,9 @@ def pull_through_add_content(self, content_artifact):
         from pulpcore.plugin.tasking import dispatch, add_and_remove
 
         body = {"repository_pk": self.pk, "add_content_units": [cpk], "remove_content_units": []}
-        return dispatch(add_and_remove, kwargs=body, exclusive_resources=[self], immediate=True)
+        task = dispatch(add_and_remove, kwargs=body, exclusive_resources=[self], immediate=True)
+        # TODO: check if task completed succesfully, as its run in the worker now
+        return task
```

Why do we return anything at all? Should this not be a dispatch and forget action?

---

Yeah, maybe it should.
I assumed that  the request would fail when the task failed because it is immediate, but actually it *may* fail or silently ignore, as it may get defered.